macros/src/lib.rs

```diff
@@ -1,17 +1,106 @@
-use std;
+use proc_macro::TokenStream;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::{quote, quote_spanned};
+use std::fmt;
+use syn::spanned::Spanned;
+
+#[cfg(any(feature = "std", test))]
+#[macro_use]
+extern crate std;
+
+#[cfg(all(not(feature = "std"), not(test)))]
+#[macro_use]
+extern crate core as std;
```

Comment:
Required to stop `shared/error.rs` and `shared/parser.rs` from complaining. I just copy-pasted it from `uuid`.